.github/workflows/build-and-test.yaml

```
@@ -124,3 +160,21 @@ jobs:
         cmake --build . --target check-emitc -- -j$(nproc)
         cmake --build . --target MLIREmitCAllTests -- -j$(nproc)
         ./unittests/MLIREmitCAllTests
+
+    - name: Cache e2e
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.E2E }}
+        key: ${{ runner.os }}-e2e-${{ hashFiles('emitc/scripts/*.py') }}
+
+    - name: Run e2e test
+      run: |
+        ./${EMITC}/build_release/bin/emitc-opt --canonicalize --inline --symbol-dce ${E2E}/model_mhlo.mlir > ${E2E}/model_canon.mlir
+        FUNCTION_NAME=$(grep -oe "@[^(]*" "${E2E}"/model_canon.mlir)
+        FUNCTION_NAME="${FUNCTION_NAME:1}"
+        sed "s/$FUNCTION_NAME/predict/g" "${E2E}"/model_canon.mlir > "${E2E}"/model_fix_name.mlir
+        ./${EMITC}/build_release/bin/emitc-opt --mhlo-control-flow-to-scf -convert-mhlo-region-ops-to-emitc --convert-mhlo-to-emitc --convert-scf-to-emitc --convert-std-to-emitc --convert-mhlo-const-to-std "${E2E}"/model_fix_name.mlir > "${E2E}"/model_emitc.mlir
```

nit. Be consistent with - and --
```suggestion
        ./${EMITC}/build_release/bin/emitc-opt --mhlo-control-flow-to-scf --convert-mhlo-region-ops-to-emitc --convert-mhlo-to-emitc --convert-scf-to-emitc --convert-std-to-emitc --convert-mhlo-const-to-std "${E2E}"/model_fix_name.mlir > "${E2E}"/model_emitc.mlir
```